kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -396,19 +397,36 @@ public final T createOrReplace(T... items) {
       return withName(itemToCreateOrReplace.getMetadata().getName()).createOrReplace(itemToCreateOrReplace);
     }
 
-    try {
-      // Create
-      KubernetesResourceUtil.setResourceVersion(itemToCreateOrReplace, null);
-      return create(itemToCreateOrReplace);
-    } catch (KubernetesClientException exception) {
-      if (exception.getCode() != HttpURLConnection.HTTP_CONFLICT) {
-        throw exception;
+    final CompletableFuture<T> future = new CompletableFuture<>();
+    while (!future.isDone()) {
+      try {
+        // Create
+        KubernetesResourceUtil.setResourceVersion(itemToCreateOrReplace, null);
+        future.complete(create(itemToCreateOrReplace));
+      } catch (KubernetesClientException exception) {
+        final T itemFromServer;
+        if (exception.getCode() == HttpURLConnection.HTTP_INTERNAL_ERROR) {
```

Comment:
I think you mentioned that 503 could also be returned in some cases.
Maybe better to check for the whole family (5xx) or a list of 5xx statuses. WDYT?